From aeafe7b4b0b798eb4e8e9a16df2fc4c5fe188f1a Mon Sep 17 00:00:00 2001
From: SunJianHao <24031212195@stu.xidian.edu.cn>
Date: Fri, 3 Apr 2026 01:59:08 +0000
Subject: [PATCH v2 1/1] smb/client: avoid null-ptr-deref when tests fail in
 test_cmp_map()

Use KUNIT_ASSERT_NOT_NULL() to abort the test cases on failure.

Reported-by: ChenXiaoSong <chenxiaosong@kylinos.cn>
Signed-off-by: SunJianHao <24031212195@stu.xidian.edu.cn>
Reviewed-by: ChenXiaoSong <chenxiaosong@kylinos.cn>
---
 fs/smb/client/smb2maperror_test.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/smb/client/smb2maperror_test.c b/fs/smb/client/smb2maperror_test.c
index 8c47dea7a2c1..8c4f168c8e99 100644
--- a/fs/smb/client/smb2maperror_test.c
+++ b/fs/smb/client/smb2maperror_test.c
@@ -21,7 +21,7 @@ test_cmp_map(struct kunit *test, const struct status_to_posix_error *expect)
 	const struct status_to_posix_error *result;
 
 	result = smb2_get_err_map_test(expect->smb2_status);
-	KUNIT_EXPECT_PTR_NE(test, NULL, result);
+	KUNIT_ASSERT_NOT_NULL(test, result);
 	KUNIT_EXPECT_EQ(test, expect->smb2_status, result->smb2_status);
 	KUNIT_EXPECT_EQ(test, expect->posix_error, result->posix_error);
 	KUNIT_EXPECT_STREQ(test, expect->status_string, result->status_string);
-- 
2.53.0

